New squeaky datum and plushes

ninjanomnom opened the pull request:
Adds a new component for making things squeak, used in:
clown shoes
mice
bike horns
plushes

Refactors carp plush to use the new /obj/item/toy/plush type

Adds new plushes

![plushies](https://puu.sh/xCemS/a58ffea692.png)

ShizCalev commented:
Donk Corporation is a member of the syndicate. Nanotrasen probably wouldn't be too cool with this idea.

praisenarsie commented:
@ShizCalev explain donk co and donkpockets then :^)

ShizCalev commented:
@praisenarsie What do you mean explain donk co? I just did.

praisenarsie commented:
yes but WHY ARE THERE DONKPOCKETS ON NT STATIONS.

ShizCalev commented:
The hold the patent, probably a generic /s

Could make a retcon PR for that if you'd like :^)

Xhuis commented:
New canon: donk pockets and Donk Co. shuttles are disliked by NT but tolerated due to their cheapness and them being better than any competition, including NT-brand shuttles.

GunHog commented:
This PR disappoints me. I thought we would have a Donksoft shuttle where the walls, floor, and interior are made entirely of foam. I was also imagining it with a bunch of Donksoft toys.

ninjanomnom commented:
get me sprites for donksoft toys and I will add some to every shuttle

praisenarsie commented:
@ShizCalev ^
make it happen ;))))))))))))))))))


(jk you don't have to)

ninjanomnom commented:
sprite request: narsie and ratvar plushes

Keekenox commented:
https://i.imgur.com/wksHFQt.png

NOW WITH SPRINGY ARM ACTION AND LIGHT UP GEMS! PEW PEW SPROING!

nicbn commented:
Make it Nanotrasen instead of Donk, Nanotrasen is the corporation that do research which allows the creation of the shuttle engines.

Keekenox commented:
Pls make PR with ratvar and narsie plushies, I will make sprites.

ninjanomnom commented:
I'm already going to do it here and add them to shuttles @Keekenox

Keekenox commented:
REEE ATOMIZE YOUR PRS AAAAAAAAAAH

Qustinnus commented:
Ninja gets lectured by keek on pr etiquette, what a fuckin day to be alive

ninjanomnom commented:
It is atomized, this is a Donk Co fluff pr relating to shuttles. Adding merchandise into the shuttles as part of it makes sense.

ninjanomnom commented:
okay *now* this may be overscoped a bit because I made a squeaky datum for making anything squeak.

It was important for plushies to put in shuttles okay

Cyberboss commented on code/datums/components/squeek.dm:
```diff
@@ -0,0 +1,24 @@
+/datum/component/squeak
+	var/list/squeak_sounds = list('sound/items/toysqueak1.ogg'=1, 'sound/items/toysqueak2.ogg'=1, 'sound/items/toysqueak3.ogg'=1)
```
make this a static list and another, initially empty list

Cyberboss commented on code/datums/components/squeek.dm:
```diff
@@ -0,0 +1,24 @@
+
+/datum/component/squeak/proc/play_squeak()
+	var/atom/owner = parent
+	playsound(owner.loc, pickweight(squeak_sounds), 30, 1, -1)
```
just playsound(parent, ...

ChangelingRain reviewed (dismissed):
holy shit atomize your pr; make the shuttle fluff change a _different_ pr

ninjanomnom commented:
yeah the shuttle change is the odd one out here though it's what all this came from lol, moment while I move it over

ninjanomnom committed:
New squeaky datum and plushies

new plush subtype
clown shoes, bike horns, and mice slightly changed to use the datum

ninjanomnom committed:
narsie and ratvar plushes

ninjanomnom commented:
@Cyberboss WIP can be removed now

optimumtact commented on code/game/objects/items/plushes.dm:
```diff
@@ -0,0 +1,42 @@
+
+/obj/item/toy/plush/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, squeak_override)
```
QDEL this on destroy?

Cyberboss replied:
No, datum/Destroy cleans up components

optimumtact commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -79,15 +80,12 @@
 	item_color = "clown"
-	var/footstep = 1	//used for squeeks whilst walking
 	pockets = /obj/item/storage/internal/pocket/shoes/clown
+	var/list/step_sounds = list('sound/effects/clownstep1.ogg'=1,'sound/effects/clownstep2.ogg'=1)
```
the step sounds should live only in the datum, not on the shoes themself I think

ninjanomnom replied:
I'll switch this to the other option which is to only do this work in initialize, but the only option other than that is to have a different component subtype for anything that wants a different squeak sound.

optimumtact replied:
hey you're the expert on this

Cyberboss commented on code/_onclick/item_attack.dm:
```diff
@@ -9,6 +9,7 @@
 // Called when the item is in the active hand, and clicked; alternately, there is an 'activate held object' verb or you can hit pagedown.
 /obj/item/proc/attack_self(mob/user)
+	SendSignal(COMSIG_ITEM_ATTACK_SELF, user)
 	return
```
Remove this trailing return

Cyberboss commented on code/datums/components/squeek.dm:
```diff
@@ -0,0 +1,59 @@
+		RegisterSignal(COMSIG_ATOM_BLOB_ACT, .proc/play_squeak)
+		RegisterSignal(COMSIG_ATOM_HULK_ATTACK, .proc/play_squeak)
+		RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/play_squeak)
+	if(istype(parent, /atom/movable))
```
Lines 29-34 can get indented again

Cyberboss commented on code/datums/components/squeek.dm:
```diff
@@ -0,0 +1,59 @@
+		RegisterSignal(COMSIG_MOVABLE_IMPACT, .proc/play_squeak)
+	else
+		RegisterSignal(COMSIG_ATOM_ENTERED, .proc/play_squeak)
+	if(istype(parent, /obj/item))
```
Should be under the atom/movable check

Cyberboss commented on code/datums/components/squeek.dm:
```diff
@@ -0,0 +1,59 @@
+		RegisterSignal(COMSIG_ITEM_ATTACK, .proc/play_squeak)
+		RegisterSignal(COMSIG_ITEM_ATTACK_SELF, .proc/use_squeak)
+		RegisterSignal(COMSIG_ITEM_ATTACK_OBJ, .proc/play_squeak)
+	if(istype(parent, /obj/item/clothing/shoes))
```
should be under the obj/iitem check

Cyberboss commented on code/game/objects/items.dm:
```diff
@@ -217,6 +217,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 	return message
 
 /obj/item/attack_self(mob/user)
+	. = ..() //Why the hell is this proc defined twice
```
Make it a single definition please

Cyberboss commented on code/game/objects/items/clown_items.dm:
```diff
@@ -107,37 +107,22 @@
 /obj/item/bikehorn/suicide_act(mob/user)
 	user.visible_message("<span class='suicide'>[user] solemnly points the horn at [user.p_their()] temple! It looks like [user.p_theyre()] trying to commit suicide!</span>")
-	playsound(src.loc, honksound, 50, 1)
+	playsound(src.loc, pickweight(honksounds), 50, 1)
```
playsound(src

ninjanomnom committed:
fixes duplicate obj/item/attack_self()

and other misc requested changes

Cyberboss commented on code/game/objects/items/clown_items.dm:
```diff
@@ -125,6 +124,10 @@
 
+/obj/item/bikehorn/airhorn/Initialize()
+	. = ..()
+	AddComponent(/datum/component/squeak, list('sound/items/airhorn2.ogg'=1), 50)
```
So, you know this will delete the component created by /obj/item/bikehorn, right?

ninjanomnom replied:
yeah, it's meant to override the old one for a new sound

Cyberboss commented:
Travis

ninjanomnom committed:
moves the sounds into initialize
